src/scraper/scraper.cpp

```diff
@@ -4708,35 +4694,19 @@ UniValue testnewsb(const UniValue& params, bool fHelp)
         res.pushKV("NewFormatSuperblock serialization", "FAILED");
     }
 
+    NewFormatSuperblock = NN::Superblock::UnpackLegacy(sSBCoreData_out);
```

Comment:
This should be NewFormatSuperblock = NN::Superblock::UnpackLegacy(sPackedSBCoreData);

I detected this because the

        try {
            m_zero_mags = std::stoi(ExtractXML(packed, "<ZERO>", "</ZERO>"));
        } catch (...) {
            LogPrintf("LegacySuperblock: Failed to parse zero mag CPIDs");
        }

was being thrown in the log.

I changed it in my local repo and the hash compare passes and this error is not thrown...

I also added

    _log(logattribute::INFO, "testnewsb", "NewFormatSuperblock legacy unpack number of zero mags = " + std::to_string(NewFormatSuperblock.m_cpids.Zeros()));

to see what was being populated in the Superblock object via the legacy unpacker.

before...

07/14/2019 01:36:45 [INFO] <testnewsb> : Checking compatibility with binary SB pack/unpack by packing then unpacking, then comparing to the original
07/14/2019 01:36:45 [INFO] <testnewsb> : Generated contract passed binary pack/unpack
07/14/2019 01:36:45 [INFO] <testnewsb> : sSBCoreData size = 40298
07/14/2019 01:36:45 [INFO] <testnewsb> : sPackedSBCoreData size = 18950
07/14/2019 01:36:45 [INFO] <testnewsb> : m_projects size = 20
07/14/2019 01:36:45 [INFO] <testnewsb> : m_cpids size = 1015
07/14/2019 01:36:45 [INFO] <testnewsb> : zero-mag count = 671
07/14/2019 01:36:45 [INFO] <testnewsb> : nNewFormatSuperblockSerSize = 17743
07/14/2019 01:36:45 [INFO] <testnewsb> : nNewFormatSuperblock_outSerSize = 17743
07/14/2019 01:36:45 [INFO] <testnewsb> : NewFormatSuperblock serialization passed.
07/14/2019 01:36:45 [INFO] <testnewsb> : NewFormatSuperblock legacy unpack number of zero mags = 0
07/14/2019 01:36:45 [INFO] <testnewsb> : NewFormatSuperblock legacy hash passed.

after...

07/14/2019 01:52:18 [INFO] <testnewsb> : Checking compatibility with binary SB pack/unpack by packing then unpacking, then comparing to the original
07/14/2019 01:52:18 [INFO] <testnewsb> : Generated contract passed binary pack/unpack
07/14/2019 01:52:18 [INFO] <testnewsb> : sSBCoreData size = 40298
07/14/2019 01:52:18 [INFO] <testnewsb> : sPackedSBCoreData size = 18950
07/14/2019 01:52:18 [INFO] <testnewsb> : m_projects size = 20
07/14/2019 01:52:18 [INFO] <testnewsb> : m_cpids size = 1015
07/14/2019 01:52:18 [INFO] <testnewsb> : zero-mag count = 671
07/14/2019 01:52:18 [INFO] <testnewsb> : nNewFormatSuperblockSerSize = 17743
07/14/2019 01:52:18 [INFO] <testnewsb> : nNewFormatSuperblock_outSerSize = 17743
07/14/2019 01:52:18 [INFO] <testnewsb> : NewFormatSuperblock serialization passed.
07/14/2019 01:52:18 [INFO] <testnewsb> : NewFormatSuperblock legacy unpack number of zero mags = 669
07/14/2019 01:52:18 [INFO] <testnewsb> : NewFormatSuperblock legacy hash passed.

Reply:
Interesting that the rounding results in a few extra zero-mag CPIDs. The new SB is filtering those out because it applies the rounding before loading the CPID/mag into the map. 

I accidentally force-pushed the branch. I don't think I rebased, so it should work with your changes. Let me know if it doesn't and I'll put these in.

Reply:
What am I saying, this is a PR... I went ahead and pushed those changes 🙂